### ISSUE TITLE
Docstring format convention

### ISSUE THREAD

**Issue by fabianegli** [nf-core/tools#1529]
### Description of feature

I think it would be great to decide on a docstring format for this project to have a chance to use them consistently throughout the codebase for documentation and the generation of user friendly API docs.

The formats I like most are the one from [Google](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) and [numpy](https://numpydoc.readthedocs.io/en/latest/format.html) with  a slight preference for the latter.

**Comment by ewels** [comment]
Yeah this one has been on my radar for a while too. I would also like to convert the Sphinx site that we have for linting docs to use MkDocs and [MkDocStrings](https://mkdocstrings.github.io/), so was hoping to do this as part of the same effort.

I think MkDocStrings only supports Google style (may be worth checking), so my vote would be for that.

**Comment by ewels** [comment]
This looks useful: https://github.com/pycqa/pydocstyle
